lib/pages/chat_list/chat_list_view_builder.dart

```diff
@@ -21,49 +25,112 @@ class ChatListViewBuilder extends StatelessWidget {
+                    room: rooms[index],
+                    chatListItem: _CommonChatListItem(
+                      controller: controller,
+                      room: rooms[index],
+                    ),
+                  )
+                : _CommonChatListItem(
+                    controller: controller,
+                    room: rooms[index],
+                  );
           },
         );
       },
     );
   }
 }
+
+class _CommonChatListItem extends StatelessWidget {
+  const _CommonChatListItem({
+    required this.controller,
+    required this.room,
+  });
+
+  final ChatListController controller;
+  final Room room;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder(
+      valueListenable: controller.widget.activeRoomIdNotifier,
+      builder: (context, activeRoomId, child) {
+        return ChatListItem(
+          room,
+          key: Key('chat_list_item_${room.id}'),
+          isEnableSelectMode: controller.isSelectMode,
+          onTap: controller.isSelectMode
+              ? () => controller.toggleSelection(room.id)
+              : null,
+          onSecondaryTap: () => controller.handleContextMenuAction(
+            context,
+            room,
+          ),
+          onLongPress: () => controller.onLongPressChatListItem(
+            room,
+          ),
+          checkBoxWidget: ValueListenableBuilder(
+            valueListenable: controller.conversationSelectionNotifier,
+            builder: (context, conversationSelection, __) {
+              final conversation = conversationSelection.firstWhereOrNull(
+                (conversation) => conversation.roomId.contains(room.id),
+              );
+              return Checkbox(
+                value: conversation?.isSelected == true,
+                onChanged: (_) {
+                  controller.toggleSelection(room.id);
+                },
+              );
+            },
+          ),
+          activeChat: activeRoomId == room.id,
+        );
+      },
+    );
+  }
+}
+
+class _SlidableChatListItem extends StatelessWidget {
+  const _SlidableChatListItem({
+    required this.controller,
+    required this.room,
+    required this.chatListItem,
+  });
+
+  final ChatListController controller;
+  final Room room;
+  final Widget chatListItem;
+
+  @override
+  Widget build(BuildContext context) {
+    return Slidable(
+      endActionPane: ActionPane(
+        motion: const ScrollMotion(),
+        extentRatio: ChatListViewStyle.slidableExtentRatio,
+        children: [
+          SlidableAction(
+            autoClose: true,
+            label: room.isFavourite
+                ? L10n.of(context)!.unpin
+                : L10n.of(context)!.pin,
+            icon: room.isFavourite ? Icons.push_pin_outlined : Icons.push_pin,
+            onPressed: (_) => controller.togglePin(room),
+            foregroundColor: ChatListViewStyle.slidableForegroundColorDefault,
+            backgroundColor: Colors.greenAccent[700] ??
```

Comment:
IMO: `Colors.greenAccent[700]` change it to m3

Reply:
Colors is already a palet of M3 since we use flutter latest version